src/mapdata.h

```diff
@@ -77,6 +77,18 @@ struct map_deconstruct_info {
     map_deconstruct_info() : can_do(false), deconstruct_above(false), ter_set(NULL_ID), furn_set(NULL_ID) {};
     bool load(JsonObject &jsobj, std::string member, bool is_furniture);
 };
+struct map_dismantle_info {
+    // Only if true, the terrain/furniture can be taken down
+    bool can_do;
+    // In the unlikely event walls can be takn down without tools.
+    bool dismantle_above;
+    // items you get when dismantling.
+    std::string drop_group;
+    ter_str_id ter_set;    // terrain to set (REQUIRED for terrain))
+    furn_str_id furn_set;    // furniture to set (only used by furniture, not terrain)
+    map_dismantle_info() : can_do(false), dismantle_above(false), ter_set(NULL_ID), furn_set(NULL_ID) {};
```

Comment:
Please use C++11 default initialization. For example `bool dismantle_above = false`. You only need to retain the constructor for `NULL_ID`